src/main/java/com/pinterest/secor/common/LogFilePath.java

```diff
@@ -145,18 +148,26 @@ public LogFilePath withPrefix(String prefix) {
     }
 
     public String getLogFileParentDir() {
-        ArrayList<String> elements = new ArrayList<String>();
+        elements = new ArrayList<String>();
         if (mPrefix != null && mPrefix.length() > 0) {
             elements.add(mPrefix);
         }
         if (mTopic != null && mTopic.length() > 0) {
-            elements.add(mTopic);
+        	if (getCustomTopicsNames() != null &&
```

Comment:
How often is this method (getLogFileParentDir()) being called?  This might be quite expensive to add 4 more calls.